backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
+    },
+]
 
-    app_configuration = AppConfigurationModel(
-        configScope=configScope,
-        version_id=version_id,
-        configuration=new_configuration,
-        changed_by=event["requestContext"]["authorizer"]["principalId"],
-        change_reason=request["changeReason"],
-        created_at=time.time(),
-    )
 
+@event_wrapper
+def update_configuration(event, context):
     try:
-        # Create a record for this configuration and mark it the latest config by incrementing the version Id
-        app_configuration_dao.create(config=app_configuration)
-    except Exception as e:
-        if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
-            e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
-        raise e
+        response_status_code = 200
+        warning_issues = []
+        request = json.loads(event["body"])
+        configScope = request["configScope"]
+        version_id = request["versionId"] + 1  # increment so this will be the latest version
+
+        new_configuration = SettingsModel.from_dict(request["configuration"])
+
+        app_configuration = AppConfigurationModel(
+            configScope=configScope,
+            version_id=version_id,
+            configuration=new_configuration,
+            changed_by=event["requestContext"]["authorizer"]["principalId"],
+            change_reason=request["changeReason"],
+            created_at=time.time(),
+        )
+
+        try:
+            # Create a record for this configuration and mark it the latest config by incrementing the version Id
+            app_configuration_dao.create(config=app_configuration)
+        except Exception as e:
+            if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
+                e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
+            raise e
 
-    return f"Successfully updated configuration for {configScope}, version {version_id}."
+        log.info(env_variables)
+        if env_variables[EnvVariable.MANAGE_IAM_ROLES] and configScope == "global":
+            # Create the new app deny policy
+            try:
+                deny_policy_statements = []
+                enabled_services_dict = app_configuration.configuration.enabled_services.to_dict()
+                log.info(f"Keys: {enabled_services_dict.items()}")
+                # Check each service
+                for service in enabled_services_dict.keys():
+                    # If the service is deactivated and has instructions on how to deny the service in 'service_disable_permissions'
+                    if not enabled_services_dict[service] and service in SERVICE_DEACTIVATE_PROPERTIES:
+                        if "ResourceType" in SERVICE_DEACTIVATE_PROPERTIES[service]:
+                            suspend_all_of_type(SERVICE_DEACTIVATE_PROPERTIES[service]["ResourceType"])
+                        if "Statements" in SERVICE_DEACTIVATE_PROPERTIES[service]:
```

Comment:
Is there ever a scenario where "Statements" wouldn't be one of the properties?

Reply:
No, but just protecting against `KeyErrors`. I guess we could remove this.